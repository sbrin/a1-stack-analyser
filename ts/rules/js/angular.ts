import { register } from '../../register.js';

register({
  tech: 'angular',
  name: 'Angular',
  type: 'language',
  files: ['angular.json'],
  dependencies: [
    { type: 'npm', name: 'angular' },
    { type: 'npm', name: /^@angular\//, example: '@angular/forms' },
  ],
});
