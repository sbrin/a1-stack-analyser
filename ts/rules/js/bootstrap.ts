import { register } from '../../register.js';

register({
  tech: 'bootstrap',
  name: 'Bootstrap',
  type: 'tool',
  dependencies: [{ type: 'npm', name: 'bootstrap' }],
});
