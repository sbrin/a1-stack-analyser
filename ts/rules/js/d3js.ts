import { register } from '../../register.js';

register({
  tech: 'd3js',
  name: 'D3.js',
  type: 'tool',
  dependencies: [
    { type: 'npm', name: 'd3' },
    { type: 'npm', name: 'd3-shape' },
    { type: 'npm', name: '@types/d3' },
    { type: 'npm', name: '@types/d3-shape' },
  ],
});
