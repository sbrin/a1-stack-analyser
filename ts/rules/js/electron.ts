import { register } from '../../register.js';

register({
  tech: 'electron',
  name: 'Electron',
  type: 'tool',
  dependencies: [
    { type: 'npm', name: 'electron' },
    { type: 'githubAction', name: 'samuelmeuli/action-electron-builder' },
  ],
});
