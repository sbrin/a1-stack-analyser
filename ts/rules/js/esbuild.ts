import { register } from '../../register.js';

register({
  tech: 'esbuild',
  name: 'Esbuild',
  type: 'tool',
  dependencies: [{ type: 'npm', name: 'esbuild' }],
});
