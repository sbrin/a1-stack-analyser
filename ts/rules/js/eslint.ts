import { register } from '../../register.js';

register({
  tech: 'eslint',
  name: 'Eslint',
  type: 'tool',
  files: ['.eslintrc', '.eslintrc.cjs', '.eslintrc.json', '.eslintrc.js'],
  dependencies: [{ type: 'npm', name: 'eslint' }],
});
