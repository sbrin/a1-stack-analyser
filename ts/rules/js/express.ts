import { register } from '../../register.js';

register({
  tech: 'express',
  name: 'Express',
  type: 'tool',
  dependencies: [{ type: 'npm', name: 'express' }],
});
