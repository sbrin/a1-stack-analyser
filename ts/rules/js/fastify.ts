import { register } from '../../register.js';

register({
  tech: 'fastify',
  name: 'Fastify',
  type: 'tool',
  dependencies: [
    { type: 'npm', name: 'fastify' },
    { type: 'npm', name: '@opentelemetry/instrumentation-fastify' },
  ],
});
