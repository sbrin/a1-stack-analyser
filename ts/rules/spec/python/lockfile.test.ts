import { describe, expect, it } from 'vitest';

import { analyser } from '../../../analyser/index.js';
import { rawList } from '../../../loader.js';
import { flatten } from '../../../payload/helpers.js';
import { FakeProvider } from '../../../provider/fake.js';

import '../../../autoload.js';

describe('python (lockfile)', () => {
  it('should match everything', async () => {
    const deps = [];
    for (const item of rawList) {
      if (item.type !== 'dependency' || item.ref.type !== 'python') {
        continue;
      }

      const example = 'example' in item.ref ? item.ref.example : item.ref.name;
      deps.push(`${example}==1.0.0`);
    }

    const lockfile: string = deps.join('\n');

    const res = await analyser({
      provider: new FakeProvider({
        paths: {
          '/': ['requirements.txt'],
        },
        files: {
          '/requirements.txt': lockfile,
        },
      }),
    });

    const merged = flatten(res, { merge: true });
    expect(Array.from(merged.techs).sort()).toMatchSnapshot();
    expect(Array.from(merged.dependencies).sort()).toMatchSnapshot();
  });

  it('should match weird syntax', async () => {
    const res = await analyser({
      provider: new FakeProvider({
        paths: {
          '/': ['requirements.txt'],
        },
        files: {
          '/requirements.txt': [
            // No version
            'tensorflow',
            // gte
            'algoliasearch>=1.9.1',
            // gte-lt
            'datadog>=1.4.3,<7.0.0',
          ].join('\n'),
        },
      }),
    });

    const merged = flatten(res, { merge: true });
    expect(Array.from(merged.techs).sort()).toMatchSnapshot();
    expect(Array.from(merged.dependencies).sort()).toMatchSnapshot();
  });
});
