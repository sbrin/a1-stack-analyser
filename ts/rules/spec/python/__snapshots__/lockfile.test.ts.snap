// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`python (lockfile) > should match everything 1`] = `
[
  "algolia",
  "celery",
  "datadog",
  "lago",
  "novu",
  "python",
  "qdrant",
]
`;

exports[`python (lockfile) > should match everything 2`] = `
[
  [
    "python",
    "algoliasearch",
    "1.0.0",
  ],
  [
    "python",
    "algoliasearch-django",
    "1.0.0",
  ],
  [
    "python",
    "celery",
    "1.0.0",
  ],
  [
    "python",
    "datadog",
    "1.0.0",
  ],
  [
    "python",
    "lago-python-client",
    "1.0.0",
  ],
  [
    "python",
    "novu",
    "1.0.0",
  ],
  [
    "python",
    "pytest-playwright",
    "1.0.0",
  ],
  [
    "python",
    "qdrant-client",
    "1.0.0",
  ],
  [
    "python",
    "selenium",
    "1.0.0",
  ],
  [
    "python",
    "tensorflow",
    "1.0.0",
  ],
]
`;

exports[`python (lockfile) > should match weird syntax 1`] = `
[
  "algolia",
  "datadog",
  "python",
]
`;

exports[`python (lockfile) > should match weird syntax 2`] = `
[
  [
    "python",
    "algoliasearch",
    "1.9.1",
  ],
  [
    "python",
    "datadog",
    "1.4.3",
  ],
  [
    "python",
    "tensorflow",
    "latest",
  ],
]
`;
